### DISCUSSION TITLE
Newly joined application not showing up on Teleport

### DISCUSSION THREAD

**Discussion by jack-bobs** [gravitational/teleport#13935] [Q&A]
I recently added a new application to my cluster.

It's not showing up on Teleport.

What am I doing wrong?

**Comment by webvictim** [comment]
Please share your `/etc/teleport.yaml` config files from the `app_service` and other relevant parts of Teleport.

**Comment by jack-bobs** [comment]
Hi, this is the config from one of our applications we'd like to join our cluster.

Just removed a few things that I don't want to share, but pretty much everything's here.

The issue I do encounter when trying to connect an app to our cluster is that when it tries to join - the process freezes at 'cache "apps" first init succeeded. cache./cache.go:708

Can anyone point out what I might be doing wrong?

```teleport:
  # Data directory for the Application Proxy service. If running on the same
  # node as Auth/Proxy service, make sure to use different data directories.
  data_dir: /var/lib/teleport-app
  # Instructs the service to load the join token from the specified file
  # during initial registration with the cluster.
  auth_token: /tmp/token
  # Proxy address to connect to. Note that it has to be the proxy address
  # because the app service always connects to the cluster over a reverse
  # tunnel.
  auth_servers:
  - [redacted]:3025
app_service:
    enabled: yes
    # Teleport provides a small debug app that can be used to make sure application
    # access is working correctly. It'll output JWTs so it can be useful
    # when extending your application.
    debug_app: true
    # This section contains definitions of all applications proxied by this
    # service. It can contain multiple items.
    apps:
      # Name of the application. Used for identification purposes.
    - name: "jtest-phishing"
      # URI and port the application is available at.
      uri: "https://0.0.0.0:3333"
      # Optional application public address to override.
auth_service:
  enabled: "yes"
ssh_service:
  enabled: "no"
proxy_service:
  enabled: "yes"
  public_addr: [redacted]```

**Reply by webvictim** [reply]
When running remotely, the `app_service` needs to join the cluster via the proxy service rather than the auth service (port 3025) so it can establish a reverse tunnel.

Change the port under `auth_servers` in your config to the proxy’s HTTPS listener instead (probably 443 or 3080) and then restart Teleport.

**Reply by webvictim** [reply]
Also, you should disable the `auth_service` and `proxy_service` in this process if your only intention is to proxy a remote app back to a central Teleport cluster.

If you are looking to run the `auth_service` and `proxy_service` in this process, you should remove the `auth_servers` entry from the config.

**Reply by webvictim** [reply]
Also, your app is probably not available on `https://0.0.0.0:3333`.

You probably mean `https://127.0.0.1:3333` if it’s listening on localhost, or something like `https://1.2.3.4:3333` if it’s listening on another IP.

**Comment by jack-bobs** [comment]
Hi @webvictim 

So, I'd only modify the application's teleport.yaml file - not the main server where the cluster is being hosted

**Reply by webvictim** [reply]
That's correct, yes. Run `app_service` only and disable the others, and connect it back to the proxy instead of the auth service. That'll work!
